clients/src/main/java/org/apache/kafka/clients/admin/TopicDescription.java

```diff
@@ -62,25 +62,25 @@ public int hashCode() {
      *                   leadership and replica information for that partition.
      */
     public TopicDescription(String name, boolean internal, List<TopicPartitionInfo> partitions) {
-        this(name, internal, partitions, Collections.emptySet());
+        this(name, Uuid.ZERO_UUID, internal, partitions);
+    }
+
+    public TopicDescription(String name, Uuid topicId, boolean internal, List<TopicPartitionInfo> partitions) {
+        this(name, topicId, internal, partitions, Collections.emptySet());
     }
 
     /**
      * Create an instance with the specified parameters.
      *
      * @param name The topic name
+     * @param topicId the topic id
      * @param internal Whether the topic is internal to Kafka
      * @param partitions A list of partitions where the index represents the partition id and the element contains
      *                   leadership and replica information for that partition.
      * @param authorizedOperations authorized operations for this topic, or null if this is not known.
      */
-    public TopicDescription(String name, boolean internal, List<TopicPartitionInfo> partitions,
+    public TopicDescription(String name, Uuid topicId, boolean internal, List<TopicPartitionInfo> partitions,
```

Comment:
We seemed to have removed a constructor?

Reply:
This is also my mistake, I tried to add a constructor but remove one, I have rollbacked the removing.